server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -188,6 +188,7 @@ static void addCollectorsAndSearch(SearchContext searchContext) throws QueryPhas
                 searchContext,
                 searchContext.parsedPostFilter() != null || searchContext.minimumScore() != null
             );
+            // TODO there is a SingleThreadCollectorManager still here that breaks with AssertionErrors once concurrency is enabled
```

Comment:
This PR doesn't trigger this problem since enabling concurrency is limited to Dfs phase, but marking this as something to discuss for later.

Reply:
I'll remove the TODO then